### PR TITLE
(feat)RDCC-2214: Fixing the ASB message.

### PR DESCRIPTION
**Pull request by nitishsingla1234**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDCC-2214

### Change description ###
Fixing the azure servicebus method.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/cwrdapi/controllers/constants/ErrorConstants.java [hunk 1]
```diff
@@ -11,14 +11,13 @@
...
 
-    CONFLICT_EXCEPTION("10 : Error was caused by duplicate key exception"),
+    CONFLICT_EXCEPTION("6 : Error was caused by duplicate key exception"),
 
-    ACCESS_EXCEPTION("9 : Access Denied"),
-    ERROR_PUBLISHING_TO_TOPIC("10 : Sorry, there is a problem with the service. Try again later"),
+    ACCESS_EXCEPTION("7 : Access Denied"),
 
-    RUNTIME_EXCEPTION("10 : Sorry, there is a problem with the service. Try again later");
+    RUNTIME_EXCEPTION("8 : Sorry, there is a problem with the service. Try again later");
```
Comment (nitishsingla1234) [inline src/main/java/uk/gov/hmcts/reform/cwrdapi/controllers/constants/ErrorConstants.java:20]: The numbers were off.

**Review by AdilOoze — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/cwrdapi/servicebus/TopicPublisher.java [hunk 2]
```diff
@@ -44,9 +44,13 @@ public void sendMessage(@NotNull Object message) {
                     ((PublishCaseWorkerData) message).getUserIds() != null
                             ? ((PublishCaseWorkerData) message).getUserIds().size() : null);
         }
-
-        jmsTemplate.convertAndSend(destination, message);
-        log.info("{}:: Message published to service bus topic", loggingComponentName);
+        try {
+            jmsTemplate.convertAndSend(destination, message);
+            log.info("{}:: Message published to service bus topic", loggingComponentName);
+        } catch (Exception ex) {
```
Comment (SujeethKumarAri) [inline src/main/java/uk/gov/hmcts/reform/cwrdapi/servicebus/TopicPublisher.java:50]: Any reason for using try catch here? As we are retrying when the exception occurs, we need not throw every time. If all the attempts fail then we already have recover method to throw the exception

Comment (nitishsingla1234) [reply src/main/java/uk/gov/hmcts/reform/cwrdapi/servicebus/TopicPublisher.java:50]: Hi Sujeeth, I agree that the recover method will come into play when any JMSException occurs. This was a suggestion from Shashank as mentioned in ticket https://tools.hmcts.net/jira/browse/RDCC-2394 to put a try catch block in  sendMessage method. Let me cross verify with him and come back to you @shashank-rastogi  FYI.

Comment (SujeethKumarAri) [reply src/main/java/uk/gov/hmcts/reform/cwrdapi/servicebus/TopicPublisher.java:50]: ah ok!! Please let me know after discussing with @shashank-rastogi . I shall approve the changes accordingly

Comment (nitishsingla1234) [reply src/main/java/uk/gov/hmcts/reform/cwrdapi/servicebus/TopicPublisher.java]: Hi @SujeethKumarAri As discussed with Shashank, we will rely on the Recover block for catching any exceptions. Removed the try/catch block.

**Review by shashank-rastogi — APPROVED** [review]

**Review by SujeethKumarAri — APPROVED** [review]

**Review by AdilOoze — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/cwrdapi/servicebus/TopicPublisher.java (hunk 2 above)
Comment (SujeethKumarAri) [reply src/main/java/uk/gov/hmcts/reform/cwrdapi/servicebus/TopicPublisher.java]: thanks @nitishsingla1234

**Pull request merged**